### PR TITLE
EN-56646: Collect query metrics

### PR DESCRIPTION
**Pull request by PaulMTT**
With this change we are timing the execution of the query-coordinator httpClient when it attempts to query data.
The intention is to collect this metric independent of which backend secondary is used.
This timing information is then used to collect a metric if things were successful, which in our case is if we "forward" the results backs up.

Function Timing.TimedResultReturningTransformed wraps a block and passes this blocks return value along with duration to a consuming function, this consuming function can apply a mapping - such as turning a result to a response.

### CODE REVIEW

File: .run/query-coordinator.run.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      <pattern>
+        <option name="PATTERN" value="com.socrata.querycoordinator.*" />
+        <option name="ENABLED" value="true" />
+      </pattern>
+    </extension>
+    <method v="2">
+      <option name="Make" enabled="true" />
+    </method>
+  </configuration>
+</component>
```
Comment (zachlefevre) [inline .run/query-coordinator.run.xml:18]: newline here

File: project/Dependencies.scala [hunk 2]
```diff
@@ -53,4 +54,6 @@ object Dependencies {
 
   val scalaTest = "org.scalatest" %% "scalatest" % versions.scalaTest % "test"
   val scalaCheck = "org.scalacheck" %% "scalacheck" % versions.scalaCheck % "test"
+
+  val rollupMetrics = "com.socrata" %% "rollup-metrics" % versions.rollupMetrics
```
Comment (zachlefevre) [inline project/Dependencies.scala:58]: This is a private repo being depended on in a public repo. I'm not sure our policy here. We may consider opening that repo.

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryExecutor.scala [hunk 3]
```diff
@@ -474,31 +477,36 @@ class QueryExecutor(httpClient: HttpClient,
...
+      Timing.TimedResultReturningTransformed{
+        using(IOUtils.toInputStream(serializedAnalyses, StandardCharsets.UTF_8.name)) { queryInputStream =>
+          val request = base.p(route).
+            addHeaders(PreconditionRenderer(precondition) ++ ifModifiedSince.map("If-Modified-Since" -> _.toHttpDate)).
+            addHeaders(extraHeaders).
+            addParameters(params).
+            blob(queryInputStream, "application/octet-stream") // blob implies POST
+          httpClient.execute(request, resourceScope)
+        }
+      }{(result,duration)=>
```
Comment (zachlefevre) [inline query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryExecutor.scala:489]: space here

File: project/Dependencies.scala (hunk 2 above)
Comment (PaulMTT) [reply project/Dependencies.scala:58]: Yeah good point, I'll definitely follow up on that.

**Pull request merged**